include/nccl_ofi_rdma.h

```diff
@@ -59,12 +59,16 @@ typedef enum nccl_net_ofi_rdma_req_type {
 	NCCL_OFI_RDMA_SEND_CONN_RESP,
 } nccl_net_ofi_rdma_req_type_t;
 
-typedef enum nccl_ofi_rdma_msg_type {
+enum nccl_ofi_rdma_msg_type {
 	NCCL_OFI_RDMA_MSG_CONN,
 	NCCL_OFI_RDMA_MSG_CONN_RESP,
 	NCCL_OFI_RDMA_MSG_CTRL,
 	NCCL_OFI_RDMA_MSG_EAGER
-} nccl_ofi_rdma_msg_type_t;
+};
+/* This goes on the wire, so we want the datatype
+ * size to be fixed.
+ */
+typedef uint16_t nccl_ofi_rdma_msg_type_t;
```

Comment:
would it be more resilient if we defined this as a bitfield instead?

Reply:
I thought about it and discussed it back then with @bwbarrett, enum implementation is compiler/platform specific and bitfield, if I am not wrong, is a hint to the compiler, it is not guaranteed that it will be respected. Since this is going on the wire we need it to be of fixed size, so `uint16_t` guarantees that.

Reply:
bitfields are fine, although you should still be careful with packing.  Assuming sizes of enums is not fine :).

Reply:
this was part of #397 that was already merged. We can re-evaluate it later if we want.